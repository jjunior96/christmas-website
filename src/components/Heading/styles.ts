import styled, { css, DefaultTheme } from 'styled-components';

import { HeadingProps } from '.';

type ContainerProps = Omit<HeadingProps, 'children'>;

const containerModifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.huge};
  `
};

export const Container = styled.h1<ContainerProps>`
  ${({ theme, size, color }) => css`
    color: ${theme.colors[color!]};

    font-weight: ${theme.font.bold};

    ${!!size && containerModifiers[size](theme)}
  `}
`;
