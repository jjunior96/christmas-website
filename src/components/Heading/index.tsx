import * as S from './styles';

export type HeadingProps = {
  size?: 'small' | 'medium' | 'large';
  color?: 'white' | 'black';
  children: React.ReactNode;
};

const Heading = ({
  size = 'large',
  color = 'white',
  children
}: HeadingProps) => {
  return (
    <S.Container size={size} color={color}>
      {children}
    </S.Container>
  );
};

export default Heading;
