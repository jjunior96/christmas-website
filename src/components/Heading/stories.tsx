import { Story, Meta } from '@storybook/react/types-6-0';

import Heading, { HeadingProps } from '.';

export default {
  title: 'Heading',
  component: Heading
} as Meta;

export const Default: Story<HeadingProps> = (args) => <Heading {...args} />;

Default.args = {
  children: 'Heading',
  color: 'black',
  size: 'large'
};
