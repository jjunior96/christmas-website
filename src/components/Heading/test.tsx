import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../utils/tests/helpers';

import theme from 'styles/theme';

import Heading from '.';

describe('<Heading />', () => {
  it('should render the heading default', () => {
    const { container } = renderWithTheme(<Heading>test</Heading>);

    expect(screen.getByRole('heading', { name: /test/i })).toBeInTheDocument();
    expect(screen.getByText(/test/i)).toHaveStyle({
      color: theme.colors.white,
      'font-size': theme.font.sizes.huge
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with color dark', () => {
    renderWithTheme(<Heading color="black">test</Heading>);

    expect(screen.getByText(/test/i)).toHaveStyle({
      color: theme.colors.black
    });
  });

  it('should render with small size', () => {
    renderWithTheme(<Heading size="small">test</Heading>);

    expect(screen.getByText(/test/i)).toHaveStyle({
      'font-size': theme.font.sizes.medium
    });
  });

  it('should render with medium size', () => {
    renderWithTheme(<Heading size="medium">test</Heading>);

    expect(screen.getByText(/test/i)).toHaveStyle({
      'font-size': theme.font.sizes.large
    });
  });
});
