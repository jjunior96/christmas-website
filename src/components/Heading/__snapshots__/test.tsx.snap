// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the heading default 1`] = `
.c0 {
  color: #fff;
  font-weight: 600;
  -webkit-letter-spacing: 4px;
  -moz-letter-spacing: 4px;
  -ms-letter-spacing: 4px;
  letter-spacing: 4px;
  font-size: 10.4rem;
}

<h1
  class="c0"
  color="white"
>
  test
</h1>
`;
